lib/nolibc/string.c

```diff
@@ -152,17 +152,21 @@ size_t strnlen(const char *str, size_t len)
 
 char *strncpy(char *dst, const char *src, size_t len)
 {
-	size_t clen;
-
-	clen = strnlen(src, len);
-	memcpy(dst, src, clen);
+	if (len != 0) {
+		char *d = dst;
+		const char *s = src;
+
+		do {
+			if ((*d++ = *s++) == 0) {
+				/* NUL pad the remaining n-1 bytes */
+				while (--len != 0)
+					*d++ = 0;
+				break;
+			}
+		} while (--len != 0);
+	}
 
-	/* instead of filling up the rest of left space with zeros,
-	 * append a termination character if we did not copy one
-	 */
-	if (clen < len && dst[clen - 1] != '\0')
-		dst[clen] = '\0';
-	return dst;
+	return (dst);
```

Comment:
Why do you need brackets here?

Reply:
oh the parenthesis are not needed, it's just common in BSD KNF, feel free to remove if wanted.
It's somewhat common in the unikraft codebase though:
sam:unikraft: grep -r return\ \( . |wc -l
388

Reply:
Okay, we'll remove the brackets while upstreaming.